[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend.

etaf opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.14.0) (oldest at bottom):
* #178130
* #174958
* #173779
* __->__ #161940

This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @chenyang78

etaf committed:
[Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined
scheduling.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


scheduling.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


scheduling.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

[ghstack-poisoned]

etaf committed:
Update on "[Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

EikanWang commented on test/inductor/test_cutlass_backend.py:
```diff
@@ -53,15 +53,24 @@
+HAS_CUDA = torch.cuda.is_available()
+HAS_GPU = HAS_CUDA or HAS_XPU
+SM90OrLater = HAS_XPU or SM90OrLater
+SM80OrLater = HAS_XPU or SM80OrLater
```
The changes conflict with the previous semantics because `SM90OrLater` and `SM80OrLater` are dedicated to CUDA. If we need to enable test cases for XPU, we should update the test case directly. For example, `@unittest.skipIf(not SM90OrLater, "need sm_90")` could be refiled as `@skipIfCUDA(not SM90OrLater, "need sm_90")`

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

EikanWang commented on test/inductor/test_cutlass_backend.py:
```diff
@@ -520,6 +539,7 @@ def forward(self, a, b):
 
             torch.testing.assert_close(actual, expected)
 
+    @skipIfXpu(msg="fp8 not supported on xpu")
```
```suggestion
    @skipIfXpu(msg="XPU has not supported fp8 yet")
```

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to CombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Switch XPU triton scheduling to combined"


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

EikanWang reviewed (commented):
Overall, it looks good to me. nit: Please refine the messages for the skipped cases.

etaf commented on test/inductor/test_cutlass_backend.py:
```diff
@@ -265,7 +277,8 @@ def test_cutlass_key(self):
         self.assertIsNotNone(cutlass_key())
 
-    @unittest.skipIf(not SM90OrLater, "need sm_90")
+    @skipIfXpu(msg="TODO: remove this after SYCL-TLA fixed accuracy issue")
```
@EikanWang , Thanks for your time. `skipCUDAIf(condition)` means skip the test if device type is cuda and `condition` match, but skipIfCuda/skipIfXpu means skip if current device is cuda/xpu. They are used for different perposes, so we would better not to make a change.

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf commented:
@pytorchbot drci

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo chenyang78

[ghstack-poisoned]

etaf committed:
Update on "[xpu][feature][Inductor XPU GEMM] Step 10/N: Enable XPU sycl-tla(Intel cutlass) backend."


This PR officially replaces the Inductor XPU backend scheduling from TritonScheduling to XPUCombinedScheduling, enabling support for both CUTLASS and Triton backends. It also refactors test_cutlass_backend to be enabled on XPU, and the sycl-tla(intel cutlass) is also added in XPU CI. Currently, the CUTLASS XPU backend does not yet support epilogue fusion.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo chenyang78

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf commented:
@pytorchbot merge

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]
